modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/spring/ByteBuddySpringCacheInstrumentator.java

```diff
@@ -0,0 +1,96 @@
+
+  /**
+   * Instrument Spring ConcurrentReferenceHashMap cache instances in order to register them into a
+   * {@link ByteBuddySpringCachesManager} that offers a cache cleanup feature.
+   * 
+   * @param targetClassloader Classloader where the Spring cache instrumentation will be injected.
+   * @see ByteBuddySpringCachesManager#clearCaches()
+   */
+  public static void instrumentForCleanup(ClassLoader targetClassloader) {
+    try {
+      TypePool targetTypePool = TypePool.Default.of(targetClassloader);
+      TypePool muleTypePool = TypePool.Default.of(ByteBuddySpringCacheInstrumentator.class.getClassLoader());
+      ByteBuddy byteBuddy = new ByteBuddy();
+      DynamicType byteBuddySpringCachesManager;
+      try (DynamicType.Unloaded<?> unloaded = byteBuddy
+          .redefine(muleTypePool.describe(BYTE_BUDDY_SPRING_CACHES_MANAGER_CLASS)
+              .resolve(),
+                    ClassFileLocator.ForClassLoader.of(ByteBuddySpringCacheInstrumentator.class.getClassLoader()))
+          .make()) {
+        byteBuddySpringCachesManager =
+            unloaded.load(targetClassloader, ClassLoadingStrategy.Default.INJECTION);
+      }
+      try (DynamicType.Unloaded<?> unloaded = byteBuddy
+          .rebase(targetTypePool.describe(CONCURRENT_REFERENCE_HASH_MAP_CLASS).resolve(),
+                  ClassFileLocator.ForClassLoader.of(targetClassloader))
+          .visit(Advice.to(byteBuddySpringCachesManager.getTypeDescription()).on(ElementMatchers.isConstructor()))
+          .make()) {
+        unloaded.load(targetClassloader, ClassLoadingStrategy.Default.INJECTION);
+      }
+      LOGGER.debug("Spring caches are now instrumented for cleanup.");
+    } catch (Exception e) {
+      LOGGER.error("Could not instrument Spring caches cleanup.", e);
+    }
+  }
+
+  /**
+   * Instrument Spring ConcurrentReferenceHashMap cache instances so that they are weakly (and not softly) referenced by default.
+   * 
+   * @param targetClassloader Classloader where the Spring cache instrumentation will be injected.
+   */
+  public static void instrumentForWeakness(ClassLoader targetClassloader) {
+    TypePool typePool = TypePool.Default.of(targetClassloader);
+    try (DynamicType.Unloaded<?> unloaded = new ByteBuddy()
+        .redefine(typePool.describe(CONCURRENT_REFERENCE_HASH_MAP_CLASS).resolve(),
+                  ClassFileLocator.ForClassLoader.of(IntrospectionUtils.class.getClassLoader()))
+        // We redirect any other constructor to the fully parameterized one.
+        .constructor(target -> target.isConstructor() && target.getParameters().size() < 4)
+        .intercept(MethodCall
+            .invoke((ElementMatcher<MethodDescription>) target -> target.isConstructor() && target.getParameters().size() == 4)
+            // We use default arguments plus WEAK reference type, discarding the intercepted ones.
+            .with(16, 0.75F, 16, WEAK))
```

Comment:
Just curiosity: Are you going to use the provided values or these hardcoded ones?